### PR TITLE
WIP: Scheduler refactor

### PR DESCRIPTION
**Pull request by BrockReece**
Just to get the ball rolling on this PR, I will be adding some more coverage for the ui actions and cleaning up a few of the component's methods but this has all the coverage for inbound and outbound data. 

This will need some careful consideration, can we please double check the snapshots match up with what is expected. I am new to this data schema and could have missed something.

### CODE REVIEW

File: src/components/shared/scheduler/SchedulerEditor.vue [hunk 1]
```diff
@@ -308,7 +153,7 @@
 
                 schedulerObject['service=scheduler;table=timetables;field=starts_at;'] = schedule.limit.startsAt ? moment(schedule.limit.startsAt).format('YYYY-MM-DD hh:mm:ss') : null
                 schedulerObject['service=scheduler;table=timetables;field=ends_at;'] = schedule.limit.endsAt ? moment(schedule.limit.endsAt).format('YYYY-MM-DD hh:mm:ss') : null
-                schedulerObject['service=scheduler;table=timetables;field=max_executions;'] = schedule.limit.maxExecutions > 0 ? parseInt(schedule.limit.maxExecutions) : null
+                schedulerObject['service=scheduler;table=timetables;field=max_executions;'] = schedule.limit.maxExecutions > 0 ? parseInt(schedule.limit.maxExecutions, 10) : null
```
Comment (matthewerskine) [inline src/components/shared/scheduler/SchedulerEditor.vue]: Does this apply a default of 10 maximum executions if `maxExecutions` cannot be parsed as an integer? If so, should we possibly apply `null` here as the default?

Comment (BrockReece) [reply src/components/shared/scheduler/SchedulerEditor.vue]: No this is the 'base' for the integer to be displayed in, base 10 is the 'decimal' numbering system. (As opposed to binary for example)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

This was actually picked up by my linter, not sure how it got this far already.

Comment (matthewerskine) [reply src/components/shared/scheduler/SchedulerEditor.vue]: Ah ok, cool stuff!

Comment (BrockReece) [reply src/components/shared/scheduler/SchedulerEditor.vue]: For simple number casting to in JS, I tend to use `Number('12')`

File: src/components/shared/scheduler/SchedulerEditor.vue [hunk 2]
```diff
@@ -238,52 +122,13 @@
             CroudDatepicker,
         },
 
+        filters: {
```
Comment (kingy04) [inline src/components/shared/scheduler/SchedulerEditor.vue]: Could this not be done with css?

**Comment by kingy04** [comment]
Great refactor using v-for loops, nice alt month logic and the tests look good to me with the schema looking correct.

File: src/components/shared/scheduler/SchedulerEditor.vue (hunk 2 above)
Comment (BrockReece) [reply src/components/shared/scheduler/SchedulerEditor.vue]: Sure, but I was refactoring whilst trying to match the existing snapshots. I can move it to a css rule if we want, but that won't be testable.

Comment (kingy04) [reply src/components/shared/scheduler/SchedulerEditor.vue]: ah, gotcha.

Comment (BrockReece) [reply src/components/shared/scheduler/SchedulerEditor.vue]: Cool, are you happy for this to be merged in?

**Comment by BrockReece** [comment]
@kingy04 are you happy for this to be merged in too?

**Pull request merged**